Return empty string if `indent` is `None` in `_indent()`

JonoYang opened the pull request:
This PR addresses the issues in https://github.com/dcbaker/jsonstreams/issues/21, where compact JSON was not being formed because `indent=0` is passed into `json.JSONEncoder` from `jsonstreams.Stream` by default, causing `json.JSONEncoder` to insert newline characters. The solution is to allow `indent` to be None when instantiating `jsonstreams.Stream` and to modify `jsonstreams.BaseWriter._indent()` to return an empty string if `indent` is None.

JonoYang committed:
Return empty string if `indent` is `None` in `_indent()`

    * Set `indent=None` instead of `indent=0` in the init args of `jsonstreams.Stream`

Signed-off-by: Jono Yang <jyang@nexb.com>

dcbaker commented on jsonstreams/__init__.py:
```diff
@@ -144,7 +144,10 @@ def __init__(self, fd, indent, baseindent, encoder, pretty):
+        if self.indent and isinstance(self.indent, int) and self.indent >= 0:
+            return ' ' * self.baseindent * self.indent
+        else:
+            return ''
```
I've really been meaning to getting around to fixing the type annotations, but let's just assume that only an int or None is valid for self.indent. In that case I think just
```python
if self.indent:
   ...
...
```
is fine.

dcbaker reviewed (changes requested):
thanks! this looks like a nice change. I've got one small style nit, and could you add a line to docs/source/changes.rst under the next heading, with the fixed issue?

JonoYang committed:
Remove additional checks on self.indent in _indent

Signed-off-by: Jono Yang <jyang@nexb.com>

JonoYang committed:
Update changelog

Signed-off-by: Jono Yang <jyang@nexb.com>

JonoYang commented:
@dcbaker I've removed the extra checks on `self.indent` in `BaseWriter._indent()` and added a new entry in the changelog. I created a new header 0.4.4 and listed the change under it.

dcbaker commented:
perfect, thanks again!
